crates/kubelet/src/kubelet.rs

```diff
@@ -36,7 +36,7 @@ use tokio::sync::mpsc;
 pub struct Kubelet<P> {
     provider: Arc<P>,
     kube_config: kube::Config,
-    config: Config,
+    config: Box<Config>,
```

Comment:
Is it necessary to take the argument as a box? I would think boxing at the beginning of the function would also work, no?

Reply:
When I tested this, I had to box it as it was coming out, but I made a few more changes since then, so I can try again. Currently my windows computer is having problems though, so I'll need someone else to test

Reply:
I think this is because of a borrow.

After removing the `Box`es from a variety of locations:

```
PS C:\Users\me\tmp\krustlet> cargo build --no-default-features --features rustls-tls
   Compiling wasi-provider v0.3.0 (C:\Users\me\tmp\krustlet\crates\wasi-provider)
   Compiling wascc-provider v0.3.0 (C:\Users\me\tmp\krustlet\crates\wascc-provider)
   Compiling krustlet v0.3.0 (C:\Users\me\tmp\krustlet)
error[E0308]: mismatched types
  --> src/krustlet-wascc.rs:24:46
   |
24 |     let provider = WasccProvider::new(store, &config, kubeconfig.clone()).await?;
   |                                              ^^^^^^^
   |                                              |
   |                                              expected struct `kubelet::config::Config`, found `&kubelet::config::Config`
   |                                              help: consider removing the borrow: `config`
```

removing the borrow resulted in the following build error:

```
PS C:\Users\me\tmp\krustlet> cargo build --no-default-features --features rustls-tls
   Compiling krustlet v0.3.0 (C:\Users\me\tmp\krustlet)
error[E0382]: use of moved value: `config`
  --> src/krustlet-wascc.rs:25:54
   |
12 |     let config = Config::new_from_file_and_flags(
   |         ------ move occurs because `config` has type `kubelet::config::Config`, which does not implement the `Copy` trait
...
24 |     let provider = WasccProvider::new(store, config, kubeconfig.clone()).await?;
   |                                              ------ value moved here
25 |     let kubelet = Kubelet::new(provider, kubeconfig, config).await?;
   |                                                      ^^^^^^ value used here after move
```

This is because the config is being borrowed by both the provider and the kubelet. There's likely some code cleanup we could do here.

In either case, cloning the config object before passing to the provider fixes the issue.

here's my diff (missed a few spots in the docstrings, but the code's all there):

```
diff --git a/crates/kubelet/src/kubelet.rs b/crates/kubelet/src/kubelet.rs
index fe0252d..0dc96c4 100644
--- a/crates/kubelet/src/kubelet.rs
+++ b/crates/kubelet/src/kubelet.rs
@@ -36,7 +36,7 @@ use tokio::sync::mpsc;
 pub struct Kubelet<P> {
     provider: Arc<P>,
     kube_config: kube::Config,
-    config: Box<Config>,
+    config: Config,
 }
 
 impl<T: 'static + Provider + Sync + Send> Kubelet<T> {
@@ -45,7 +45,7 @@ impl<T: 'static + Provider + Sync + Send> Kubelet<T> {
     pub async fn new(
         provider: T,
         kube_config: kube::Config,
-        config: Box<Config>,
+        config: Config,
     ) -> anyhow::Result<Self> {
         Ok(Self {
             provider: Arc::new(provider),
diff --git a/crates/wascc-provider/src/lib.rs b/crates/wascc-provider/src/lib.rs
index 6b1ebc0..012ae36 100644
--- a/crates/wascc-provider/src/lib.rs
+++ b/crates/wascc-provider/src/lib.rs
@@ -147,7 +147,7 @@ impl WasccProvider {
     /// (including creating it if necessary)
     pub async fn new(
         store: Arc<dyn Store + Sync + Send>,
-        config: &Box<kubelet::config::Config>,
+        config: kubelet::config::Config,
         kubeconfig: kube::Config,
     ) -> anyhow::Result<Self> {
         let host = Arc::new(Mutex::new(WasccHost::new()));
diff --git a/src/krustlet-wascc.rs b/src/krustlet-wascc.rs
index 8dd4a29..8043408 100644
--- a/src/krustlet-wascc.rs
+++ b/src/krustlet-wascc.rs
@@ -9,10 +9,10 @@ use wascc_provider::WasccProvider;
 async fn main() -> anyhow::Result<()> {
     // The provider is responsible for all the "back end" logic. If you are creating
     // a new Kubelet, all you need to implement is a provider.
-    let config = Box::new(Config::new_from_file_and_flags(
+    let config = Config::new_from_file_and_flags(
         env!("CARGO_PKG_VERSION"),
         None,
-    ));
+    );
 
     // Initialize the logger
     env_logger::init();
@@ -21,7 +21,7 @@ async fn main() -> anyhow::Result<()> {
 
     let store = make_store(&config);
 
-    let provider = WasccProvider::new(store, &config, kubeconfig.clone()).await?;
+    let provider = WasccProvider::new(store, config.clone(), kubeconfig.clone()).await?;
     let kubelet = Kubelet::new(provider, kubeconfig, config).await?;
     kubelet.start().await
 }
diff --git a/src/krustlet-wasi.rs b/src/krustlet-wasi.rs
index 8266ef1..5b8d67d 100644
--- a/src/krustlet-wasi.rs
+++ b/src/krustlet-wasi.rs
@@ -9,10 +9,10 @@ use wasi_provider::WasiProvider;
 async fn main() -> anyhow::Result<()> {
     // The provider is responsible for all the "back end" logic. If you are creating
     // a new Kubelet, all you need to implement is a provider.
-    let config = Box::new(Config::new_from_file_and_flags(
+    let config = Config::new_from_file_and_flags(
         env!("CARGO_PKG_VERSION"),
         None,
-    ));
+    );
 
     // Initialize the logger
     env_logger::init();
```

Reply:
Ok, I double checked this as well and the box is no longer needed. It has been removed